src/renderer/services/slippi/slippi.service.ts

```diff
@@ -25,45 +29,54 @@ const handleErrors = (errors: readonly GraphQLError[] | undefined) => {
 };
 
 class SlippiBackendClient implements SlippiBackendService {
-  private httpLink: HttpLink;
   private client: ApolloClient<NormalizedCacheObject>;
 
   constructor(
     private readonly authService: AuthService,
     private readonly dolphinService: DolphinService,
     clientVersion?: string,
   ) {
-    this.httpLink = new HttpLink({ uri: SLIPPI_BACKEND_URL });
-    this.client = new ApolloClient({
-      link: this.httpLink,
-      cache: new InMemoryCache(),
-      name: "slippi-launcher",
-      version: clientVersion,
-    });
-  }
+    this.authService = authService;
+    const httpLink = new HttpLink({ uri: SLIPPI_BACKEND_URL });
+    const authLink = setContext(async () => {
+      // The firebase ID token expires after 1 hour so we will update the header on all actions
+      const token = await this.authService.getUserToken();
 
-  // The firebase ID token expires after 1 hour so we will refresh it for actions that require it.
-  private async _refreshAuthToken(): Promise<AuthUser> {
-    const user = this.authService.getCurrentUser();
-    if (!user) {
-      throw new Error("User is not logged in.");
-    }
-    const token = await this.authService.getUserToken();
-
-    const authLink = new ApolloLink((operation, forward) => {
-      // Use the setContext method to set the HTTP headers.
-      operation.setContext({
+      return {
         headers: {
-          authorization: token ? `Bearer ${token}` : "",
+          authorization: token ? `Bearer ${token}` : undefined,
         },
-      });
-
-      // Call the next link in the middleware chain.
-      return forward(operation);
+      };
+    });
+    const retryLink = new RetryLink({
+      delay: {
+        initial: 300,
+        max: Infinity,
+        jitter: true,
+      },
+      attempts: {
+        max: 3,
```

Comment:
How come this one retries 3 times but the other one retries 5 times? Should we make it consistent?

Reply:
whoops, meant to make them both 3